packages/core/src/package-graph/__tests__/package-graph.spec.ts

```diff
@@ -19,6 +38,30 @@ describe('PackageGraph', () => {
       expect(() => new PackageGraph(pkgs)).toThrowError('Package name "pkg-2" used in multiple packages:\n\t/test/pkg-2\n\t/test/pkg-3');
     });
 
+    it('can read catalog & catalogs from "pnpm-workspace.yaml" file', () => {
+      const pkgs = [];
+      const graph = new PackageGraph(pkgs);
+
+      expect(graph.readWorkspaceCatalogConfig()).toEqual({
+        catalog: {
+          'package-1': '2.3.4',
+          'fs-extra': '^11.2.0',
+          'p-map': '^7.0.3',
+          tinyrainbow: '^2.0.0',
+        },
+        catalogs: {
+          react17: {
+            react: '^17.0.2',
+            'react-dom': '^17.0.2',
+          },
+          react18: {
+            react: '^18.2.0',
+            'react-dom': '^18.2.0',
+          },
+        },
+      });
+    });
```

Comment:
need an npa test that will verify named catalog are pulled correctly in npa `fetchSpec` variable